core/load-git.c

```diff
@@ -834,6 +836,10 @@ static void parse_dc_event(char *line, struct membuffer *str, struct git_parser_
 		line = parse_keyvalue_entry(parse_event_keyvalue, &p, line, str);
 	}
 
+	/* Only modechange events should have a divemode - fix up any corrupted names */
+	if (p.has_divemode && strcmp(p.name, "modechange"))
+		p.name = "modechange";
```

Comment:
after fixing things up we should mark the dive list as modified and make sure that the cache for this dive is invalidated. Otherwise this fix only gets written out if the user changes anything else in that dive...

Reply:
My thinking was that it didn't matter on disk, since we have the code to fix it up anyway.

The important part is that it's fixed up in our data structures, so that we show the event correctly etc.

We have other places where we do somewhat similar fixups when loading. In fact, some of them are in that same function (look for the "Older logs might mark the dive to be CCR by having .." comment just below).

There are actually many more subtle changes we do, where we might save in a different format than we loaded in (because we have various "don't save if the value hasn't changed" heuristics etc, and those heuristics have changed over the years).

So I don't think we need to mark it dirty.

But it certainly wouldn't be *wrong* to mark it dirty, so I'm not against it. It's just that I don't really expect anybody to read the git save format directly outside of subsurface and care about it...

Reply:
that's a valid point. I was thinking about this as a "get the stored format correct" exercise, but in reality that's not relevant. What matters is that we can read what's on disk and do the right thing with it.